### DISCUSSION TITLE
WebService for MySQL database

### DISCUSSION THREAD

**Discussion by ghost** [community/community#21506] [Other Feature Feedback, Questions, & Ideas]
<p>Hi,</p>
<p>I want to create a WebService which communicates with my MySQL database, the data of the database should be displayed on a Xamarin App later on. But at first I got stuck at a point of my WebService and don´t really know how to get further (I´m kinda new to programming and MySQL). Here is a piece of my code (the code is from personPersistance class):</p>
<pre><code>using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using SimpleRESTServer.Models;
using MySql.Data;

namespace SimpleRESTServer
{
    public class PersonPersistance
    {

        private MySql.Data.MySqlClient.MySqlConnection conn;

        public PersonPersistance()
        {
            string myConnectionString;
            myConnectionString = "server=127.0.0.1;uid=Local Instance MySQL80; pwd=;database=employeedb";
            try
            {
                conn = new MySql.Data.MySqlClient.MySqlConnection();
                conn.ConnectionString = myConnectionString;
                conn.Open();
            }
            catch (MySql.Data.MySqlClient.MySqlException ex)
            {

            }

        }
        public long savePerson(Person personToSave)
        {
            String sqlString = " INSERT INTO tblpersonnel (FirstName, LastName) VALUES ('"+ personToSave.FirstName + "','" + personToSave.LastName + "')";}
            MySql.Data.MySqlClient.MySqlCommand cmd = new MySql.Data.MySqlClient.MySqlConnection(sqlString, conn);
            cmd.ExecuteNonQuery();
            long id = cmd.LastInsertedId;
            return id;
        }

    }
}
</code></pre>
<p>I used this tutorial for my WebService: <a href="https://youtu.be/LpySuvYPMZQ" rel="nofollow noopener">https://youtu.be/LpySuvYPMZQ</a></p>
<p>Everthing works as it should till I come to the savePerson function. In my database I insert the ID manually, in the tutorial it gets inserted automatically (I think). So my problem is that I don´t really know how to go on at this point.</p>
<p>Here is a picture of my database columns it may help:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/6/6910efbb43278d39dddf33e28634a223a3561aa0.png" data-download-href="/uploads/short-url/eZsq4H7VlwlT77wpTbxFzgtqjkY.png?dl=1" title="StackOverflow.PNG" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/10137/181065045-757613dc-f063-4343-958d-4c9a73ca89a8.png" alt="StackOverflow.PNG" data-base62-sha1="eZsq4H7VlwlT77wpTbxFzgtqjkY" width="690" height="84" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/6/6910efbb43278d39dddf33e28634a223a3561aa0_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">StackOverflow.PNG</span><span class="informations">967×119 7.56 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Thanks in advance!</p>

**Comment by fire-eggs** [comment, marked as answer]
<p>Numeric ids can often be “auto generated” in databases. In the case of MySQL, you define the column as AUTO_INCREMENT to get this capability. [This would be my personal choice.]</p>
<p>If you want to continue using the manual method, you have two choices:</p>
<ol>
<li>
<p>Track the person id in the caller of your web service, and provide the id to use to the SavePerson function.</p>
</li>
<li>
<p>In SavePerson, query the database first to find out what the next id would be, then provide that id value when you write to the database. E.g. if you did a “select max(ID) from tblPersonnel” you get the largest ID used, then add one and include it in the INSERT statement. Note that you may end up re-using IDs [e.g. add person A with id 10; delete person A; add person B - person B will end up with ID 10].</p>
</li>
</ol>

**Comment by ghost** [comment]
<p>Thank you I will go for the auto increment option. I think it has more advantages (didn´t knew how to use it before). The code I mentioned above should work if I use auto increment in the DB, or?</p>

**Comment by ghost** [comment]
<p>Managed to do it thanks for your help!</p>

**Comment by fire-eggs** [comment]
<p>Glad to hear it! If you could use the “mark as solution”, it would be appreciated by myself and future users …</p>
